MELA/interface/TEvtProb.hh

```diff
@@ -97,7 +97,9 @@ public:
   // Reset the IO record, called at te beginning of each comoputation
   void ResetIORecord();
 
-  double XsecCalc_XVV();
+  double XsecCalc_XVV(
+    int nhel=-1
```

Comment:
I think the question but forgot whether a good answer was provided. Wouldn't a default of -1 be confusing? What are other allowed values?

Reply:
A value of -1 for the helicity indicates that you would like your matrix element to sum over all helicities, which is why it is the default. Any other value indicates a desired helicity value.

Reply:
This is the setup for the input to the Madgraph calculation function, `smatrixhel`.

Reply:
input is as follows:
nhel >= 0 -> that specific helicity
nhel = -1 -> sum over helicities
nhel < -1 -> invalid, goes to 0

Reply:
Helicity can be positive, negative, or 0 (depending on what the spin quantum number is). What is the correspondence between MG numbering convention and physics?

Reply:
I looked deeper into this, and the helicity specification seems to be indexing all the different helicity combinations that are possible.

it looks something like `DATA (NHEL(I,  23),I=1,6) /-1, 1, 1, 1,-1,-1/` (2 gluons and 4 leptons), and is dependent upon the final state. I'm honestly not entirely sure what we would do to ensure that people know what helicity corresponds to what value. 

I'm hesitant to remove the helicity calculation options altogether, as it may be useful to someone in the future. Perhaps we can make an appendix to some sort of manual that details all the possible combinations?

Reply:
Is there a way to look at the LHE record of helicities when you generate a few events with different nhel options?

Reply:
I'm not sure what you mean by this - nhel isn't a generation option. 

There is a list of all the possible helicitiies (however many there are - in ggF for this case there are 64). They are numbered from 1 to 64, each of them corresponding to a different setup like shown above (with nhel=23). When nhel=-1, all the possible contributions are summed over to make something equivalent to what MELA in MCFM and JHUGen does. When a specific nhel is selected, it only looks at the contributions from that specific final state of helicities.

I've already asked on the MG@NLO forums how to access the helicity configuration for each "index" without looking at the code, if that's possible.

Reply:
I'm not sure what you mean by this - nhel isn't a generation option. 

There is a list of all the possible helicitiies (however many there are - in ggF for this case there are 64). They are numbered from 1 to 64, each of them corresponding to a different setup like shown above (with nhel=23). When nhel=-1, all the possible contributions are summed over to make something equivalent to what MELA in MCFM and JHUGen does. When a specific nhel is selected, it only looks at the contributions from that specific final state of helicities.

I've already asked on the MG@NLO forums how to access the helicity configuration for each "index" without looking at the code, if that's possible.

Reply:
You can see the question here:
https://answers.launchpad.net/mg5amcnlo/+question/817453

Reply:
Thanks, Mohit. I suspect the different combinations are just the signs of the helicities of the 6 particles.

Assuming this is the case, let's first check which sign corresponds to left- vs. right-handedness while MG experts take the time to respond. Are you able to modify Z couplings? If so, can you set the right- or left-handed Z->ll couplings to 0 manually and check which nhel configurations give a nonzero ME in an arbitrary set of Higgs couplings?

Reply:
I have answers from MG experts on this topic. So, I have confirmed that we have access to the information contained within the helicities, and a way to find them. I also have information from the original [HELAS paper](https://inspirehep.net/literature/336604) that shows the helicity relationship. The image is for fermions, but the point still stands.

![image](https://github.com/JHUGen/JHUGenMELA/assets/15961349/620b52f5-26bc-4bf4-8ccc-06ccdb139279)


Helicities are also recorded in LHE files for things generated with Madgraph - but JHUGen/MCFM does not do this - we just report all of them as the same number.

My main point (and I'm going to go away from my original opinion here) is that I don't actually think we need the helicity stuff anyways - it seems irrelevant for our use cases (as seen by the fact that we ignore it completely when we do our own Monte Carlo!).

This is my opinion as of right now - and I think we should move forwards with a simple -1 in the nhel category always, removing the option from computeP. In the future, if necessary, we can integrate and revisit it, but I have doubts we ever will.

Reply:
> Helicities are also recorded in LHE files for things generated with Madgraph - but JHUGen/MCFM does not do this - we just report all of them as the same number.

Correct. This is why I was asking this information in the first place, and to understand the exact relationship between the LHE record and the MG convention.

I think in general having access to exclusive helicity states is valuable; this is something I hoped could also be done for JHUGen MEs in the future as well (if you think about a massive parallelization scheme using GPUs etc., there is absolutely no reason why the Fortran code should be doing the summation over different helicity configurations - that is a lower-level job that is suitable for GPUs, with interface through C++ code - not Fortran!). It is a matter of preparing the input record for MEs rather than a ME property itself - if the function that calls the ME knows how to recognize the helicity information, it should be able to pick it up without requiring the user to do anything extra in the function call. In that sense, it might be more general to implement helicity information directly into the MELAParticle class, e.g., with an additional char (I don't think any of our particles will have |m_s/s|>127).

I would leave it to you to decide whether you would like to incorporate this information in such a way, or postpone for another time in case you have other time considerations.

Reply:
I think that for now, seeing no imminent use case, I will default to a matrix element that is the result of summing over many of them.